architecture/src/androidLibMain/kotlin/navigation/Navigator.kt

```diff
@@ -139,29 +141,44 @@ class ActivityNavigator<A : NavigationAction<*>>(private val navigationMapper: (
         transaction.commit()
     }
 
-    private fun removeFragment(removeFragmentSpec: NavigationSpec.RemoveFragment) {
+    private fun removeFragment(removeFragmentSpec: NavigationSpec.RemoveFragment, bundle: NavigationBundle<*>?) {
         assert(manager != null)
-        val fragmentManager = removeFragmentSpec.getFragmentManager(manager!!)
+        val fragmentManager = removeFragmentSpec.getFragmentManager(manager!!).apply {
+            removeFragmentSpec.fragmentRequestKey?.let {
+                if (bundle != null)
```

Comment:
Maybe we need to bump ktlint to enforce brackets for single line statement as well :)